dadjokes-onchain/next/src/app/page.jsx

```diff
@@ -0,0 +1,50 @@
+"use client";
+
+import { useState } from "react";
+import Card from "@/components/Card";
+import ConnectWallet from "@/components/ConnectWallet";
+import { Button } from "@/components/Button";
+import { useJokes } from "@/hooks/useJokes";
+
+const Page = () => {
+  const [currentJokeIndex, setCurrentJokeIndex] = useState(0);
+  const jokes = useJokes();
+
+  const handlePreviousJoke = () => {
+    setCurrentJokeIndex((prevIndex) =>
```

Comment:
this naming may be a bit confusing. React passes in the current Idx as the value of prevIndex right?  if my understanding/memory on useState callback-based updates  is correct then shouldnt prevIndex be `currentIndex` and the functions need to be renamed accordingly?  Something for you to consider.

Reply:
`prevIndex` references the 'previous index' before clicking the button. We are changing the index to what is now the current index. I can see it going either way. Current before we update so previous?